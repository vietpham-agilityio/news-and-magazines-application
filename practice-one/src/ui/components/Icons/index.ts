export * from '@/ui/components/Icons/User';
export * from '@/ui/components/Icons/Plus';
export * from '@/ui/components/Icons/Home';
export * from '@/ui/components/Icons/File';
export * from '@/ui/components/Icons/Close';
export * from '@/ui/components/Icons/Circle';
export * from '@/ui/components/Icons/Search';
export * from '@/ui/components/Icons/SaveTag';
export * from '@/ui/components/Icons/MailBox';
export * from '@/ui/components/Icons/Twitter';
export * from '@/ui/components/Icons/ChatBox';
export * from '@/ui/components/Icons/Ellipse';
export * from '@/ui/components/Icons/Schedule';
export * from '@/ui/components/Icons/ArrowTop';
export * from '@/ui/components/Icons/ThreeDots';
export * from '@/ui/components/Icons/Triangle';
export * from '@/ui/components/Icons/Instagram';
export * from '@/ui/components/Icons/BurgerMenu';
export * from '@/ui/components/Icons/ChevronsUp';
export * from '@/ui/components/Icons/ArrowLeft';
export * from '@/ui/components/Icons/ArrowBack';
export * from '@/ui/components/Icons/ArrowRight';
export * from '@/ui/components/Icons/PaperPlane';
export * from '@/ui/components/Icons/ArrowBottom';
export * from '@/ui/components/Icons/ChatBoxSmall';
export * from '@/ui/components/Icons/SaveTagSmall';