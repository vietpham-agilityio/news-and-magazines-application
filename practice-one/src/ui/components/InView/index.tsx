'use client';

import { InView } from 'react-intersection-observer';

interface IProps {
  children: React.ReactNode;
}

const InViewWrapper = ({ children }: IProps) => {
  return (
    <InView triggerOnce threshold={1}>
      {({ ref, inView }) => (
        <div ref={ref} className="w-full">
          {inView && children}
        </div>
      )}
    </InView>
  );
};

export default InViewWrapper;
