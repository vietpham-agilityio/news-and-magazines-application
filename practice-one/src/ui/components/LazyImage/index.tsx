// types
import { CSSProperties } from 'react';

// components
import Image, { StaticImageData } from 'next/image';

// utils
import generateRGBDataURL from '@/utils/color';

interface IProps {
  alt: string;
  fill?: boolean;
  priority?: boolean;
  sizes?: string;
  width?: number;
  height?: number;
  objectFit?: string;
  style?: CSSProperties;
  additionalClasses?: string;
  src: string | StaticImageData;
}

export const LazyImage = ({
  src,
  alt,
  width,
  height,
  sizes='',
  style = {},
  fill = false,
  objectFit = '',
  priority = false,
  additionalClasses = '',
}: IProps) => (
  <Image
    src={src}
    alt={alt}
    fill={fill}
    sizes={sizes}
    width={width}
    style={style}
    height={height}
    placeholder="blur"
    priority={priority}
    objectFit={objectFit}
    className={additionalClasses}
    blurDataURL={generateRGBDataURL(64, 64, 64)}
  />
);
