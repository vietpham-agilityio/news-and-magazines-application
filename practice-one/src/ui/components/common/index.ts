export * from '@/ui/components/common/Tag';
export * from '@/ui/components/common/Menu';
export * from '@/ui/components/common/Title';
export * from '@/ui/components/common/Input';
export * from '@/ui/components/common/Avatar';
export * from '@/ui/components/common/Button';
export * from '@/ui/components/common/BoxIcon';
export * from '@/ui/components/common/SubMenu';
export * from '@/ui/components/common/TextArea';
export * from '@/ui/components/common/Typography';
export * from '@/ui/components/common/Information';
