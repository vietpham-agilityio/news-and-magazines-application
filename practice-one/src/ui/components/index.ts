// common
export * from '@/ui/components/common';

// shared
export * from '@/ui/components/Brand';
export * from '@/ui/components/NavItem';
export * from '@/ui/components/Comment';
export * from '@/ui/components/Carousel';
export * from '@/ui/components/SearchBar';
export * from '@/ui/components/PaperPost';
export * from '@/ui/components/BurgerMenu';
export * from '@/ui/components/Pagination';
export * from '@/ui/components/CommentForm';
export * from '@/ui/components/BreadCrumbs';
export * from '@/ui/components/TitleSection';
export * from '@/ui/components/SubmitButton';
export * from '@/ui/components/Card/CardSingle';
export * from '@/ui/components/Skeleton/Comment';
export * from '@/ui/components/Skeleton/CardPost';
export * from '@/ui/components/Skeleton/BadgeGroup';
export * from '@/ui/components/Skeleton/DetailPost';
export * from '@/ui/components/Skeleton/CardVertical';
export * from '@/ui/components/Skeleton/CardCategory';
export { default as CardPost } from '@/ui/components/Card/CardPost';
export { default as CardCategory } from '@/ui/components/Card/CardCategory';