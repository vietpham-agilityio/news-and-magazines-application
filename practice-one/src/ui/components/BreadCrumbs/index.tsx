// components
import Link from 'next/link';
import { Typography } from '@/ui/components';

// types
import { FontWeight, IBreadCrumbItem, Size } from '@/types';

// icons
import { ArrowRight } from '@/ui/components/Icons';

interface IProps {
  listBreadCrumb: IBreadCrumbItem[];
}

export const BreadCrumbs = ({ listBreadCrumb }: IProps) => (
  <div className="flex items-center">
    {listBreadCrumb.map((item, index) => {
      const { path, name } = item;

      return (
        <div key={name} className="flex items-center">
          <Link href={path}>
            <Typography
              tag="h2"
              textSize={Size.XS}
              weight={FontWeight.Normal}
              additionalClasses={`
              ${
                index === listBreadCrumb.length - 1
                  ? 'w-57.5 text-dark-90 h-5 truncate text-ellipsis overflow-hidden sm:w-full'
                  : ' text-dark-100 hover:text-primary-100'
              }`}
            >
              {name}
            </Typography>
          </Link>
          {index !== listBreadCrumb.length - 1 && (
            <i className="mx-2">
              <ArrowRight />
            </i>
          )}
        </div>
      );
    })}
  </div>
);
