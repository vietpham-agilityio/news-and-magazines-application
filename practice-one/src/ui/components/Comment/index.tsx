// static image
import ImageStore from '@/constants/images';

// components
import { Avatar, Button, Typography } from '@/ui/components';

// icons
import { ArrowBack, Schedule } from '@/ui/components/Icons';

// types
import { FontWeight, Size } from '@/types';

interface IProps {
  userName: string;
  content: string;
  publicDate?: string;
}

export const Comment = ({ userName, content, publicDate }: IProps) => (
  <div className="bg-white-90 p-4 rounded-xl">
    { publicDate ? (
      <div className="flex flex-1 px-1 mb-4">
        <Avatar
          width={60}
          height={60}
          url={ImageStore.AvatarUser}
          alt="Author this comment"
        />
        <div className="flex flex-1 items-center justify-between">
          <div className="flex flex-1 justify-between">
            <div className="flex flex-col h-full ml-2.5">
              <Typography
                tag="h2"
                textSize={Size.SM}
                additionalClasses="text-dark-100"
              >
                {userName}
              </Typography>
              <div className="flex gap-2">
                <Schedule />
                <Typography
                  tag="h3"
                  weight={FontWeight.Medium}
                  additionalClasses="text-dark-75 text-p leading-5"
                >
                  {publicDate}
                </Typography>
              </div>
            </div>
          </div>
        </div>
      </div>
    ) : (
      <Typography tag="h2" textSize={Size.SM} additionalClasses="text-dark-100">
        {userName}
      </Typography>
    )}
    <Typography
      tag="h3"
      weight={FontWeight.Light}
      additionalClasses={`${publicDate ? 'text-xsm text-black-75 leading-5' : 'text-p text-dark-75'}`}
    >
      {content}
    </Typography>
  </div>
);
