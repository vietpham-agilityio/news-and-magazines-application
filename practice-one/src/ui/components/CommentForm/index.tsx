// components
import { createComment } from '@/services';
import { Input, SubmitButton, TextArea } from '@/ui/components';

// icons
import { ChatBoxSmall } from '@/ui/components/Icons';

interface IProps {
  postId: string;
}

export const CommentForm = ({postId}: IProps) => {

return (
  <form action={createComment}>
    <div className="grid grid-cols-12 grid-rows-3 lg:gap-6">
      <div className="row-span-3 col-span-12 lg:col-span-6">
        <input name="postId" type="hidden" value={postId} />
        <div className="mb-5 row-span-1">
          <Input label="Name" name="name" type="text" />
        </div>
        <div className="mb-5 row-span-1">
          <Input label="Email" name="email" type="email" />
        </div>
        <div className="mb-5 row-span-1">
          <Input label="Website" name="website" type="text" />
        </div>
      </div>
      <div className="row-span-3 col-span-12 lg:col-span-6">
        <TextArea name='content' label="Comment" placeholder="What do you think?" />
      </div>
    </div>
    <div className="flex w-full justify-end mt-7 mb-30 sm:mt-10 sm:mb-12 lg:mt-0">
      <SubmitButton
        label="Send Comment"
        leftIcon={<ChatBoxSmall />}
      />
    </div>
  </form>
)}
