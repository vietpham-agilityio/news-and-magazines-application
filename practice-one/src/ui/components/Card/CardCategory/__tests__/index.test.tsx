import { render, waitFor, screen } from '@testing-library/react';

// service
import { getAuthorById, getPostDataById } from '@/services';

// component
import { CardCategory } from '@/ui/components';
import { mockAuthorData, mockPostDetailData } from '@/mocks';

jest.mock('../../../../../services', () => ({
  ...jest.requireActual('../../../../../services'),
  getAuthorById: jest.fn(),
  getPostDataById: jest.fn(),
}));

beforeEach(() => {
  (getAuthorById as jest.MockedFunction<typeof Object>).mockResolvedValue(
    mockAuthorData
  );
  (getPostDataById as jest.MockedFunction<typeof Object>).mockResolvedValue(
    mockPostDetailData
  );
});

afterEach(() => {
  jest.clearAllMocks();
});

describe('CardCategory component', () => {
  test('Should render match with snapshot.', async () => {
    const { container } = render(
      await CardCategory({
        id: "2"
      })
    );

    await waitFor(() => {
      expect(container).toMatchSnapshot();
      expect(screen.getByText('Benzamine')).toBeInTheDocument();
      expect(screen.getByAltText('author this post')).toBeInTheDocument();
      expect(screen.getByText('Go Home After Long Time')).toBeInTheDocument();
      expect(screen.getByText("Download torrents from verified or trusted uploaders. If you're a BitTorrent user looking for safety tips, use this method. Both of the big-name BitTorrent indexers (The Pirate Bay and KickAssTorrents) use symbols to highlight torrents uploaded by verified users.")).toBeInTheDocument();
    });
  });
});
