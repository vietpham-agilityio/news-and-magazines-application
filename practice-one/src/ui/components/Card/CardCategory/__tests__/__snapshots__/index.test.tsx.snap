// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardCategory component Should render match with snapshot. 1`] = `
<div>
  <a
    href="/articles/2"
  >
    <div
      class="mx-auto bg-white-100 flex flex-col items-center justify-center p-2.5 shadow-custom rounded-xl hover:cursor-pointer hover:opacity-90 w-89.5 h-[385px] sm:w-full max-w-186 sm:h-full sm:flex-row sm:gap-2.5 lg:flex-col lg:w-90 lg:h-[389px]"
    >
      <div
        class="relative w-84.5 h-46.5 sm:w-47.5 lg:w-85 lg:h-47.5"
      >
        <img
          alt="Go Home After Long Time"
          class="rounded-xl"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="(max-width: 603px) 338px, (min-width: 604px) and (max-width: 1023px) 190px, (min-width: 1024px) 340px"
          src="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1716223996696-ad6252a7fe7c%3Fw%3D500%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1yZWxhdGVkfDF8fHxlbnwwfHx8fHw%253D&w=3840&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1716223996696-ad6252a7fe7c%3Fw%3D500%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1yZWxhdGVkfDF8fHxlbnwwfHx8fHw%253D&w=16&q=75 16w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1716223996696-ad6252a7fe7c%3Fw%3D500%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1yZWxhdGVkfDF8fHxlbnwwfHx8fHw%253D&w=32&q=75 32w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1716223996696-ad6252a7fe7c%3Fw%3D500%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1yZWxhdGVkfDF8fHxlbnwwfHx8fHw%253D&w=48&q=75 48w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1716223996696-ad6252a7fe7c%3Fw%3D500%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1yZWxhdGVkfDF8fHxlbnwwfHx8fHw%253D&w=64&q=75 64w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1716223996696-ad6252a7fe7c%3Fw%3D500%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1yZWxhdGVkfDF8fHxlbnwwfHx8fHw%253D&w=96&q=75 96w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1716223996696-ad6252a7fe7c%3Fw%3D500%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1yZWxhdGVkfDF8fHxlbnwwfHx8fHw%253D&w=128&q=75 128w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1716223996696-ad6252a7fe7c%3Fw%3D500%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1yZWxhdGVkfDF8fHxlbnwwfHx8fHw%253D&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1716223996696-ad6252a7fe7c%3Fw%3D500%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1yZWxhdGVkfDF8fHxlbnwwfHx8fHw%253D&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1716223996696-ad6252a7fe7c%3Fw%3D500%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1yZWxhdGVkfDF8fHxlbnwwfHx8fHw%253D&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1716223996696-ad6252a7fe7c%3Fw%3D500%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1yZWxhdGVkfDF8fHxlbnwwfHx8fHw%253D&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1716223996696-ad6252a7fe7c%3Fw%3D500%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1yZWxhdGVkfDF8fHxlbnwwfHx8fHw%253D&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1716223996696-ad6252a7fe7c%3Fw%3D500%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1yZWxhdGVkfDF8fHxlbnwwfHx8fHw%253D&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1716223996696-ad6252a7fe7c%3Fw%3D500%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1yZWxhdGVkfDF8fHxlbnwwfHx8fHw%253D&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1716223996696-ad6252a7fe7c%3Fw%3D500%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1yZWxhdGVkfDF8fHxlbnwwfHx8fHw%253D&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1716223996696-ad6252a7fe7c%3Fw%3D500%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1yZWxhdGVkfDF8fHxlbnwwfHx8fHw%253D&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1716223996696-ad6252a7fe7c%3Fw%3D500%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1yZWxhdGVkfDF8fHxlbnwwfHx8fHw%253D&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
        />
      </div>
      <div
        class="flex flex-col flex-1"
      >
        <div
          class="flex flex-col gap-4 p-4"
        >
          <h2
            aria-level="2"
            class="text-sm font-medium text-dark-100 text-ellipsis overflow-hidden"
            role="heading"
          >
            Go Home After Long Time
          </h2>
          <h3
            aria-level="3"
            class="text-xsm font-light text-dark-75 h-10 w-82 leading-5 text-ellipsis overflow-hidden"
            role="heading"
          >
            Download torrents from verified or trusted uploaders. If you're a BitTorrent user looking for safety tips, use this method. Both of the big-name BitTorrent indexers (The Pirate Bay and KickAssTorrents) use symbols to highlight torrents uploaded by verified users.
          </h3>
        </div>
        <div
          class="flex flex-1 bg-white-90 rounded-xl px-4 py-3.5 mx-2.5"
        >
          <div
            style="width: 44px; height: 44px; position: relative;"
          >
            <img
              alt="author this post"
              class="rounded-xl"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="44px"
              src="/_next/image?url=https%3A%2F%2Fwww.unsplash.com%2Fimage%3Fv&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fwww.unsplash.com%2Fimage%3Fv&w=16&q=75 16w, /_next/image?url=https%3A%2F%2Fwww.unsplash.com%2Fimage%3Fv&w=32&q=75 32w, /_next/image?url=https%3A%2F%2Fwww.unsplash.com%2Fimage%3Fv&w=48&q=75 48w, /_next/image?url=https%3A%2F%2Fwww.unsplash.com%2Fimage%3Fv&w=64&q=75 64w, /_next/image?url=https%3A%2F%2Fwww.unsplash.com%2Fimage%3Fv&w=96&q=75 96w, /_next/image?url=https%3A%2F%2Fwww.unsplash.com%2Fimage%3Fv&w=128&q=75 128w, /_next/image?url=https%3A%2F%2Fwww.unsplash.com%2Fimage%3Fv&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fwww.unsplash.com%2Fimage%3Fv&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fwww.unsplash.com%2Fimage%3Fv&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fwww.unsplash.com%2Fimage%3Fv&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fwww.unsplash.com%2Fimage%3Fv&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fwww.unsplash.com%2Fimage%3Fv&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fwww.unsplash.com%2Fimage%3Fv&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fwww.unsplash.com%2Fimage%3Fv&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fwww.unsplash.com%2Fimage%3Fv&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fwww.unsplash.com%2Fimage%3Fv&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
            />
          </div>
          <div
            class="flex flex-1 items-center justify-between"
          >
            <div
              class="flex flex-col ml-2.5"
            >
              <h4
                aria-level="4"
                class="text-xsm font-medium text-dark-100"
                role="heading"
              >
                Benzamine
              </h4>
              <h5
                aria-level="5"
                class="text-md font-light text-dark-75 text-p leading-5"
                role="heading"
              >
                2024-5-30
              </h5>
            </div>
            <svg
              fill="none"
              height="21"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13.625 0C14.641 0 15.5.86 15.5 1.875V18.75c0 .977-1.055 1.563-1.914 1.094L8 16.562l-5.625 3.282C1.515 20.312.5 19.727.5 18.75V1.875C.5.859 1.32 0 2.375 0h11.25Zm0 17.656V2.11c0-.117-.117-.234-.273-.234H2.57c-.117 0-.195.117-.195.234v15.547L8 14.375l5.625 3.281Z"
                fill="#999598"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </a>
</div>
`;
